service/history/ndc/workflow.go

```diff
@@ -222,6 +222,12 @@ func (r *WorkflowImpl) FlushBufferedEvents() error {
 	}
 
 	_, err = r.failWorkflowTask(lastWriteVersion)
+	if _, err := r.mutableState.AddWorkflowTaskScheduledEvent(
```

Comment:
nit: this looks like util function just for flushing and caller should decide if a new workflow task is needed or not. It's possible that we may want to call this func to flush buffer before say terminating a workflow (I know it's not the case today)?

Reply:
caller should decide whether to create the workflow task or not, and this is already handled by FlushBufferedEvents and terminateWorkflow.